indent rule can break on arrayExpression inside a CallExpression

* **ESLint Version:** 3.12.0
* **Node Version:** 7.2.1
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
no configuration
<!-- Paste your configuration below: -->
```




```

**What did you do? Please include the actual source code causing the issue.**

Running this:

```sh
eslint --no-eslintrc --rule='indent: [2, 2, {CallExpression: {arguments: first}, ArrayExpression: first}]' test.js
```

on file `test.js`:

<!-- Paste the source code below: -->
```js
echo = spawn('cmd.exe',
             ['/c', 'echo', 'hello&&', 'echo',
              'node&&', 'echo', 'and&&', 'echo', 'world']);





```

**What did you expect to happen?**

No error

**What actually happened? Please include the actual, raw output from ESLint.**

```console
/Users/trott/test/test.js
  2:14  error  Expected indentation of 2 spaces but found 13  indent

✖ 1 problem (1 error, 0 warnings)
```

--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -722,16 +722,28 @@ module.exports = {
                 const parent = node.parent;
 
                 nodeIndent = getNodeIndent(parent).goodChar;
-                if (parentVarNode && parentVarNode.loc.start.line !== node.loc.start.line) {
+                if (!parentVarNode || parentVarNode.loc.start.line !== node.loc.start.line) {
                     if (parent.type !== "VariableDeclarator" || parentVarNode === parentVarNode.parent.declarations[0]) {
                         if (parent.type === "VariableDeclarator" && parentVarNode.loc.start.line === parent.loc.start.line) {
                             nodeIndent = nodeIndent + (indentSize * options.VariableDeclarator[parentVarNode.parent.kind]);
                         } else if (parent.type === "ObjectExpression" || parent.type === "ArrayExpression") {
-                            if (typeof options[parent.type] === "number") {
+                            const parentElements = node.parent.type === "ObjectExpression" ? node.parent.properties : node.parent.elements;
+
+                            if (parentElements[0].loc.start.line === parent.loc.start.line && parentElements[0].loc.end.line !== parent.loc.start.line) {
+
+                                /*
+                                 * If the first element of the array spans multiple lines, don't increase the expected indentation of the rest.
+                                 * e.g. [{
+                                 *        foo: 1
+                                 *      },
+                                 *      {
+                                 *        bar: 1
+                                 *      }]
+                                 * the second object is not indented.
+                                 */
+                            } else if (typeof options[parent.type] === "number") {
                                 nodeIndent += options[parent.type] * indentSize;
                             } else {
-                                const parentElements = node.parent.type === "ObjectExpression" ? node.parent.properties : node.parent.elements;
-
                                 nodeIndent = parentElements[0].loc.start.column;
                             }
                         } else if (parent.type === "CallExpression" || parent.type === "NewExpression") {

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -1907,6 +1907,13 @@ ruleTester.run("indent", rule, {
             "\t);\n" +
             "}",
             options: ["tab"]
+        },
+        {
+            code:
+            "echo = spawn('cmd.exe',\n" +
+            "             ['foo', 'bar',\n" +
+            "              'baz']);",
+            options: [2, {ArrayExpression: "first", CallExpression: {arguments: "first"}}]
         }
     ],
     invalid: [
@@ -3849,6 +3856,18 @@ ruleTester.run("indent", rule, {
             "}",
             options: [2, {ArrayExpression: 4}],
             errors: expectedErrors([2, 2, 4, "ExpressionStatement"])
+        },
+        {
+            code:
+            "echo = spawn('cmd.exe',\n" +
+            "            ['foo', 'bar',\n" +
+            "             'baz']);",
+            output:
+            "echo = spawn('cmd.exe',\n" +
+            "             ['foo', 'bar',\n" +
+            "             'baz']);",
+            options: [2, {ArrayExpression: "first", CallExpression: {arguments: "first"}}],
+            errors: expectedErrors([2, 13, 12, "ArrayExpression"])
         },
     ]
 });

